### PR TITLE
added Doomsday article

### PR DESCRIPTION
**Pull request by HawkPlays06**

### CODE REVIEW

File: content/blog/doomsday-rule/index.md [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+    authors: ['SkyHawk_0'],
+    tags: ['Python', 'Maths', 'Doomsday rule'],
+    attached: [],
+    license: 'cc-by-4'
+}
+---
+
+# What is the doomsday rule?
+
+Before I get to how my program works, I should probably explain what the doomsday rule is. The doomsday rule is an algorithm that determines the day of the week for a given date in the Gregorian calendar, e.g. the 18th of March 1898 is a Friday. I can't think when you would use this in everyday life but it's a cool party trick to show off with.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:15]:
```suggestion
Before I get to how my program works, I should probably explain what the doomsday rule is. The doomsday rule is an algorithm that determines the day of the week for a given date in the Gregorian calendar. For example, without looking at a calendar, I know that the 18th of March, 1898 is a Friday. I can't think when you would use this in everyday life, but it's a cool party trick to show off with.
```

Minor grammar change (adding commas), which lead to a minor rewrite to help with flow.

File: content/blog/doomsday-rule/index.md [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+
+The algorithm was invented by John Conway in 1973. You may know some of his work like Conway's game of life. A nice little fact about the dates that Conway was born on and died on were doomsdays but I will tell what they are later. The algorithm is simple enough that it can be calculated mentally. Conway could do this in a couple of seconds, and to improve his speed he created a program which gave him ten random dates every time he logged on to his computer.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:17]:
```suggestion
The algorithm was invented by John Conway in 1973. You may know some of his work, like Conway's game of life. Here's a nice little fact: the dates that Conway was born and died on were both doomsdays. We'll explain what doomsdays are later. The algorithm is simple enough that it can be calculated mentally. Conway could do this in a couple of seconds, and to improve his speed, he created a program that gave him ten random dates every time he logged on to his computer.
```

Rewrote the "nice fact" sentence to read a little bit smoother. It was difficult for me to follow originally, but that **is** a fun fact!

File: content/blog/doomsday-rule/index.md [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+
+# things you should know before the numbers
+
+I gathered this information from the Wikipedia page which can be found [Here](https://en.wikipedia.org/wiki/Doomsday_rule)._
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:21]:
```suggestion
I gathered this information from the [Wikipedia page, which can be found here](https://en.wikipedia.org/wiki/Doomsday_rule).
```

File: content/blog/doomsday-rule/index.md [hunk 4]
```diff
@@ -0,0 +1,660 @@
...
+
+## What is MOD?
+
+If you didn't know already when you get the MOD of something you divide the first number by the second number but the answer is the remainder. For example, 70 MOD 12 is 10, as 70 / 12 is 5 with a remainder of 10.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:25]:
```suggestion
If you didn't know already, when you get the MOD of something you divide the first number by the second number, but the answer is the remainder. For example, 70 MOD 12 is 10, as 70 / 12 is 5 with a remainder of 10.
```

File: content/blog/doomsday-rule/index.md [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+```
+This makes it very easy to add numbers to them.
+
+## Anchor Days
+
+The Wikipedia page doesn't actually explain what an anchor day actually is, but it helps work out what the doomsday is for that year.
+
+## Doomsdays
+
+The rule is called Doomsday Rule for a reason, that reason is it relies on days that are called doomsdays.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:47]:
```suggestion
The rule is called Doomsday Rule for a reason; that reason is it relies on days that are called doomsdays.
```

File: content/blog/doomsday-rule/index.md [hunk 6]
```diff
@@ -0,0 +1,660 @@
...
+
+I have mentioned these a lot, but what are they? They are dates of a year that fall on the same day of the week.
+
+So examples of this, going back to the 18th of March 1898 as an example, the 4th of April is a Monday, so is the 6th of June, same with 8th of August, 10th of October and 12th of December. There is one for every month, but I have only listed the 4th, 6th, 8th, 10th and 12th month, so this leaves the January, February, March, May, July, September and November, or the 1st, 2nd, 3rd, 5th, 7th, 9th and 11th month. I say the numbers and will from now on as it helps with the calculations later on.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:51]: This first sentence is a little long and a bit hard to follow as a result. What if we rewrote it like so?

```suggestion
For examples of this, let's use the 18th of March 1898 for reference. To start, let's note that all of the following dates fall on a Monday:

- 4th of April
- 6th of June
- 8th of August
- 10th of October 
- 12th of December

There is one for every month, but I have only listed the 4th, 6th, 8th, 10th, and 12th months; This leaves January, February, March, May, July, September, and November, or the 1st, 2nd, 3rd, 5th, 7th, 9th and 11th month. I say the numbers and will from now on as it helps with the calculations later on.
```

File: content/blog/doomsday-rule/index.md [hunk 7]
```diff
@@ -0,0 +1,660 @@
...
+
+For the 1st month, it depends if it's a leap year or not, to work this out you divide the year by 4 and if it is a whole number it's a leap year. If it's a leap year it's the 4th, if it's not it's the 3rd. Easy way of remembering this is if it's the fourth year (so a leap year) it's the 4th. So for the 18th of March 1898, as 1898 is not divisible by 4. The doomsday in January is the 3rd.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:53]:
```suggestion
For the 1st month, it depends if it's a leap year or not. To work this out, you divide the year by 4, and if it is a whole number, it's a leap year. So if it's a leap year, it's the 4th; if it's not, it's the 3rd. An easy way of remembering this is if it's the fourth year (so a leap year), it's the 4th. So for the 18th of March 1898, as 1898 is not divisible by 4. Because of this, the doomsday in January is the 3rd.
```

Going great so far!

I thought this paragraph would gain a lot from some tiny transition statements, what do ya think? :)

File: content/blog/doomsday-rule/index.md [hunk 8]
```diff
@@ -0,0 +1,660 @@
...
+
+For the 2nd month, it also depends if it's a leap year, this time it's always the last day of the month. So if it was a leap year it would be the 29th, but if it wasn't it would be the 28th. So like in the paragraph above, because 1898 is not divisible by 4, the doomsday in February is the 28th.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:55]:
```suggestion
For the 2nd month, it also depends if it's a leap year; this time, the doomsday is always the last day of the month. So if it were a leap year, it would be the 29th, but if it weren't, it would be the 28th. So like in the paragraph above, because 1898 is not divisible by 4, the doomsday in February is the 28th.
```

It wasn't quite clear what "it" meant here so I added "the doomsday" to bring some context back

File: content/blog/doomsday-rule/index.md [hunk 9]
```diff
@@ -0,0 +1,660 @@
...
+
+For the 3rd month its Pi day, yep, Pi day is a doomsday. So using the example, Pi day is a Monday for the 18th of March 1898.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:57]:
```suggestion
For the 3rd month, the doomsday is Pi day. Yep, Pi day is a doomsday. So using the example, Pi day is a Monday for the 18th of March 1898.
```

File: content/blog/doomsday-rule/index.md [hunk 10]
```diff
@@ -0,0 +1,660 @@
...
+
+For the 5th month the 9th is a doomsday, an easy way of remembering it is, "working 9 to 5...", the dots are there as there is another bit to the mnemonic but that is for another month. As you may have noticed (if you write your dates with the month first then day, I'm British, so for me it's day then month) the mnemonic is the wrong way round, this is because it works the other way round, for the 9th month the 5th is a doomsday.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:59]:
```suggestion
For the 5th month, the 9th is a doomsday. An easy way of remembering it is "working 9 to 5". This mnemonic has a second half to it, but that's for another month. As you may have noticed (if you write your dates with the month first then day, I'm British, so for me, it's day then month), the mnemonic is the wrong way round; this is because it works the other way round as well, for the 9th month the 5th is a doomsday.
```

I thought the "mnemonic" sentence was a bit hard to follow, so I made a minor change to improve it IMO

File: content/blog/doomsday-rule/index.md [hunk 11]
```diff
@@ -0,0 +1,660 @@
...
+
+Now it’s just the 7th month and the 11th month. This is where the rest of the mnemonic comes. For the 7th month, the 11th is a doomsday, and the same the other way, for the 11th month, the 7th is a doomsday. So the full mnemonic is "working 9 to 5, at 7-11"
+
+List of all doomsdays (written is day/month):   
+`3/1 OR 4/1, 28/2 OR 29/2, 14/3, 4/4, 5/9, 6/6, 7/11, 8/8, 9/5, 10/10, 11/7, 12/12`
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:64]:
```suggestion

- 3/1 or 4/1
- 28/2 or 29/2
- 14/3
- 4/4
- 5/9
- 6/6
- 7/11
- 8/8
- 9/5
- 10/10
- 11/7
- 12/12
```

Might be easier to read as a list

**Review by crutchcorn — CHANGES_REQUESTED** [review]
I'm pausing here - I'll resume my edits tomorrow - but this is looking awesome so far! 

Just some suggestions and grammar fixes

File: content/blog/doomsday-rule/index.md (hunk 10 above)
Comment (HawkPlays06) [reply content/blog/doomsday-rule/index.md:67]: Yeah, that makes it so much easier to follow thanks.

File: content/blog/doomsday-rule/index.md [hunk 12]
```diff
@@ -0,0 +1,680 @@
...
+
+# The fun part, Coding!
+
+## The Solver
+
+I wrote the script using functions so that it would be easy to edit the code.
+
+### Coding part 1 *The backbone of this project*
+
+First I thought it would good to start with the harder bit, which is the actual solver. So what is the first part you need for the calculations? Inputs. The script needs three inputs, day, month and year we will call them their respectable names, so it will look like this:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:153]:
```suggestion
First I thought it would be good to start with the harder bit, which is the actual solver; But what is the first part you need for the calculations? Inputs. Let's build a script to generate those inputs.

The script needs to output three inputs: a day, month, and year. We will call them their respectable names, so it will look like this:
```

File: content/blog/doomsday-rule/index.md [hunk 13]
```diff
@@ -0,0 +1,680 @@
...
+
+```python
+day = int(input("What day do you want? (number needed) "))
+month = int(input("What month do you want? (number needed) "))
+year = int(input("What year do you want? "))
+```
+
+We put `int` there to tell the program that whatever the input is, it's an integer not a string.
+
+So now we need the script to work out the anchor day for the century. So whilst I was making the script whenever I tried to do `century MOD 4` it would be wrong, but on the Wikipedia it has a different way of working out the anchor day which is:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:163]:
```suggestion
Now we need the script to work out the anchor day for the century. Whilst I was making the script whenever I tried to do `century MOD 4` it would be wrong, but on Wikipedia, it has a different way of working out the anchor day which is:
```

File: content/blog/doomsday-rule/index.md [hunk 14]
```diff
@@ -0,0 +1,680 @@
...
+
+`5 * (Century MOD 4) MOD 7 + 2`
+
+The way you work out the century is:
+
+`year / 100`
+
+but this usually, unless the year ended in 00, would be a decimal, so in code we write it as:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:171]:
```suggestion
However, unless the year ended in 00, this output would be a decimal, but we need a non-decimal number. So in code we write it as:
```

**Review by crutchcorn — CHANGES_REQUESTED** [review]
Heads up - you can merge multiple of these together using the "add suggestion to batch" button in the "changes" overview.

I'll be finishing this review tonight :)

File: content/blog/doomsday-rule/index.md [hunk 15]
```diff
@@ -0,0 +1,685 @@
...
+
+## The Solver
+
+I wrote the script using functions so that it would be easy to edit the code.
+
+### Coding part 1 *The backbone of this project*
+
+First I thought it would be good to start with the harder bit, which is the actual solver; But what is the first part you need for the calculations? Inputs. Let's build a script to generate those inputs.
+
+The script needs to output three inputs: a day, month, and year. We will call them their respectable names, so it will look like this:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:158]:
```suggestion
The script needs to output three inputs: a day, month, and year. We will call them their respective names, so it will look like this:
```

File: content/blog/doomsday-rule/index.md [hunk 16]
```diff
@@ -0,0 +1,685 @@
...
+
+```python
+day = int(input("What day do you want? (number needed) "))
+month = int(input("What month do you want? (number needed) "))
+year = int(input("What year do you want? "))
+```
+
+We put `int` there to tell the program that whatever the input is, it's an integer not a string.
+
+Now we need the script to work out the anchor day for the century. Whilst I was making the script whenever I tried to do `century MOD 4` it would be wrong, but on Wikipedia, it has a different way of working out the anchor day which is:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:168]:
```suggestion
Now we need the script to work out the anchor day for the century. Unfortunately, while I was making the script, whenever I tried to do the `century MOD 4` calculation, it would be incorrect. Luckily, Wikipedia has a different way of working out the anchor day, which is:
```

File: content/blog/doomsday-rule/index.md [hunk 17]
```diff
@@ -0,0 +1,685 @@
...
+
+`5 * (Century MOD 4) MOD 7 + 2`
+
+The way you work out the century is:
+
+`year / 100`
+
+However, unless the year ended in 00, this output would be a decimal, but we need a non-decimal number. So in code we write it as:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:176]:
```suggestion
Unless the year ended in 00, the output of this division would be a decimal, but we need a non-decimal number. To solve for this, we can write the following in our code:
```

File: content/blog/doomsday-rule/index.md [hunk 18]
```diff
@@ -0,0 +1,685 @@
...
+`int(year / 100)`
+
+So under the inputs we put:
+
+```python
+Century = int(year / 100)
+
+anchor = 5 * (Century % 4) % 7 + 2
+```
+(the percentage sign (%) is python's MOD)
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:187]:
```suggestion

> The percentage sign (%) is python's MOD
```

This is markdown's "_aside_" feature, which suits this well

File: content/blog/doomsday-rule/index.md [hunk 19]
```diff
@@ -0,0 +1,685 @@
...
+
+The program now can get the anchor day. The next step is for it to work out the doomsday for the year. But we have not told the program how to get the last 2 digits of the year, so we tell the script to convert it to a string, then tell it to get the 3rd and 4th digit. so this would look like this:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:189]:
```suggestion
The program now can get the anchor day. The next step is to work out the doomsday for the year. To do this, we must tell the program how to get the last 2 digits of the year. We'll tell the script to convert the year to a string, then tell it to get the 3rd and 4th character. That code looks like this:
```

File: content/blog/doomsday-rule/index.md [hunk 20]
```diff
@@ -0,0 +1,685 @@
+---
+{
+    title: "Doomsday Rule",
+    description: 'In this blog I talk about the Doomsday Rule, how it works, how to put it into code then how to make a program that tests you.',
+    published: '2019-06-29T22:12:03.284Z',
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:5]:
```suggestion
    published: '2022-01-11T22:12:03.284Z',
```

File: content/blog/doomsday-rule/index.md [hunk 21]
```diff
@@ -0,0 +1,685 @@
...
+    authors: ['SkyHawk_0'],
+    tags: ['Python', 'Maths', 'Doomsday rule'],
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:7]:
```suggestion
    tags: ['python', 'math'],
```

File: content/blog/doomsday-rule/index.md [hunk 22]
```diff
@@ -0,0 +1,685 @@
...
+B = int(EndTwo) % 12
+
+C = int(B / 4)
+
+Doomsday = ((A + B + C) % 7) + anchor
+
+Doomsday = Doomsday % 7
+```
+
+Now we need the script to find the doomsday that’s in the month chosen. But the script doesn't know what a doomsday is, so we will create a 2D list which contains the doomsday for every month. We will put the list just after the inputs to keep the script structured. The list will be written as:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:242]:
```suggestion
Now we need the script to find the doomsday in the chosen month. But the script doesn't know what a doomsday is, so we will create a 2D list that contains the doomsday every month. We will put the list just after the inputs to keep the script structured. The list will be written as:

```

File: content/blog/doomsday-rule/index.md [hunk 23]
```diff
@@ -0,0 +1,685 @@
...
+```python
+DoomsdayList = [[4, 4], [6, 6], [8, 8], [10, 10], [12, 12], [9, 5], [5, 9], [14, 3], [7, 11], [11, 7], [4, 1], [29, 2]]
+```
+
+So now the script knows what the doomsdays are, we can make a loop that loops for how many items are in the list, 12, then if `month` is the same as `a` (from the loop) but as it is a 2D list we also need to tell it to look in the second column, so it would look like this:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:247]:
```suggestion
Now that the script knows what the doomsdays are, loop through all 12 items in the list. Then we can check if `month` is the same as `a` (declared in the loop). However, since it is a 2D list, we also need to tell it to look in the second column. As a result, the code looks like this:
```

File: content/blog/doomsday-rule/index.md [hunk 24]
```diff
@@ -0,0 +1,685 @@
...
+
+```python
+for a in range(len(DoomsdayList)):
+    if DoomsdayList[a][1] == month:
+        location = a
+```
+But this causes a problem, `DoomsdayList` has only the dates for a leap year. But what if it's not a leap year? We just have a if statement after saying what `DoomsdayList` is that is if `year` MOD 4 is greater than or equal to 1, set the 10th item of the list to `[3, 1]` and the 11th item to `[28, 2]`. This would look like:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:254]:
```suggestion
But this causes a problem; `DoomsdayList` has only the dates for a leap year. But what if it's not a leap year? We can solve this problem by adding an if statement after our check. This if statement should check if `year MOD 4` is greater than or equal to 1. If this statement is true, set the 10th item of the list to `[3, 1]` and the 11th item to `[28, 2]`.
```

File: content/blog/doomsday-rule/index.md [hunk 25]
```diff
@@ -0,0 +1,685 @@
...
+
+```python
+if year % 4 >= 1:
+    DoomsdayList[10] = [3, 1]
+    DoomsdayList[11] = [28, 2]
+```
+
+### Quick summary
+Just surmising what we have done so far, we have 3 inputs: year, month and day. The script calculates the anchor day for the century. It gets the last two digits of the year then uses the a, b and c calculations. Then it adds them, gets the MOD 7 of that then adds the anchor. Then it gets the MOD 7 of that answer. The answer for the last MOD is the doomsday. We have a `DoomsdayList` which contains the doomsdays. We have an if statement that changes the 10th and 11th item if it’s not a leap year. Then we have a loop that looks for the location of the doomsday in the picked month.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:263]:
```suggestion
Just summarizing what we have done so far, we have three inputs: year, month, and day. The script calculates the anchor day for the century. It gets the last two digits of the year then uses the a, b and c calculations. Once this is done, it adds them together, gets the MOD 7 of that then adds the anchor. Next, it receives the MOD 7 of that answer. The answer for the last MOD is the doomsday. We have a `DoomsdayList` which contains the doomsdays. We have an if statement that changes the 10th and 11th item if it's not a leap year. Then we have a loop that looks for the location of the doomsday in the picked month.
```

Reads a bit better

File: content/blog/doomsday-rule/index.md [hunk 26]
```diff
@@ -0,0 +1,685 @@
...
+
+Doomsday = Doomsday % 7
+
+for a in range(len(DoomsdayList)):
+    if DoomsdayList[a][1] == month:
+        location = a
+```
+
+### Coding part 2 *Doomsday and printing the right day of the week*
+The most difficult bit is now done. We now only need to worry about the day of the month. The script now knows the location of the closest doomsday. But sort of, we haven't told it to get the item in that location, which is pretty easy:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:300]:
```suggestion

The most difficult bit is now done. Now, we only need to worry about the day of the month. After all, the script now knows the location of the closest doomsday. But this is only half true; we haven't told it to get the item in that location, which is luckily pretty easy to do:

```

File: content/blog/doomsday-rule/index.md [hunk 27]
```diff
@@ -0,0 +1,685 @@
...
+DayOfWeek = Doomsday + difference
+
+DayOfWeek = DayOfWeek % 7
+
+print("This date falls on a", weekList[DayOfWeek])
+```
+
+## The Tester
+
+With the hardest bit now out of the way and put into functions we can now relax. Make a new file but keep the old when as you can look back at it to see how it works and even modify it, we built it so that it can be used to understand and can be used as a template for further programs that will use it, like the tester.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:395]:
```suggestion
With the hardest bit now out of the way and put into functions, we can now relax. Make a new file but keep the old when as you can look back at it to see how it works and even modify it. We built it to be used to understand what our logic should do and can be used as a template for other programs that will use it: like the tester.
```

File: content/blog/doomsday-rule/index.md [hunk 28]
```diff
@@ -0,0 +1,685 @@
...
+
+I built the tester using `randint` from the module `random`. this should be your first line of code:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:397]:
```suggestion
I built the tester using `randint` from the module `random`. This should be your first line of code:
```

File: content/blog/doomsday-rule/index.md [hunk 29]
```diff
@@ -0,0 +1,685 @@
...
+```python
+from random import randint
+```
+### Coding part 1: *The functions*
+So in the previous program we made 3 inputs Year, Month, Day. We still need to give these to the program, but we need it to do it randomly, and to keep the structure we will use functions.
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:402]:
```suggestion

In the previous program, we made three inputs: Year, Month, Day. We still need to pass these to the doomsday program, but we need to do it randomly. To keep our code's structure, we will use functions.
```

File: content/blog/doomsday-rule/index.md [hunk 30]
```diff
@@ -0,0 +1,685 @@
...
+
+#### Random Year
+To get a random year we will just make a function that has two arguments, `StartYear` and `EndYear`. These will be in a `randint` line:
```
Comment (crutchcorn) [inline content/blog/doomsday-rule/index.md:405]:
```suggestion

To get a random year, we will just make a function that has two arguments: `StartYear` and `EndYear`. These will be in a `randint` line:

```

**Review by crutchcorn — COMMENTED** [review]
This is the final review pass. After this we can publish :)
